cuqi/model/_model.py

```diff
@@ -457,8 +460,130 @@ def __len__(self):
 
     def __repr__(self) -> str:
         return "CUQI {}: {} -> {}.\n    Forward parameters: {}.".format(self.__class__.__name__,self.domain_geometry,self.range_geometry,cuqi.utilities.get_non_default_args(self))
+
+
+class AffineModel(Model):
+    """ Model representing an affine operator, i.e. a linear operator with a fixed shift.
+
+    Parameters
+    ----------
+
+    linear_operator : 2d ndarray or callable function.
+        The linear operator. If ndarray is given, the operator is assumed to be a matrix.
+
+    shift : scalar or array_like
+        The shift to be added to the forward operator.
+
+    linear_operator_adjoint : callable function, optional
+        The adjoint of the linear operator. Used for computing gradients.
+
+    range_geometry : cuqi.geometry.Geometry
+        The geometry representing the range.
+
+    domain_geometry : cuqi.geometry.Geometry
+        The geometry representing the domain.
+
+    Methods
+    -----------
+    :meth:`forward` the forward operator.
+    :meth:`get_matrix` returns an ndarray with the matrix representing the forward operator.
+
+    """
+
+    def __init__(self, linear_operator, shift, linear_operator_adjoint=None, range_geometry=None, domain_geometry=None):
+
+        if not callable(linear_operator):
+            forward_func = lambda x: self._matrix@x + shift
+            forward_func_noshift = lambda x: self._matrix@x
+            adjoint_func_noshift = lambda y: self._matrix.T@y
+            matrix = linear_operator
+        else:
+            forward_func = lambda *args, **kwargs: linear_operator(*args, **kwargs) + shift
+            forward_func_noshift = linear_operator
+            adjoint_func_noshift = linear_operator_adjoint
+            matrix = None
+
+        #Check if input is callable
+        if not callable(adjoint_func_noshift):
+            raise TypeError("Adjoint of linear operator must be defined as a callable function of some kind")
+
+        # Use matrix to derive range_geometry and domain_geometry
+        if matrix is not None:
+            if range_geometry is None:
+                range_geometry = _DefaultGeometry1D(grid=matrix.shape[0])
+            if domain_geometry is None:
+                domain_geometry = _DefaultGeometry1D(grid=matrix.shape[1])  
+
+        #Initialize Model class
+        super().__init__(forward_func, range_geometry, domain_geometry)
+
```

Comment:
```suggestion
       
        # This is ensures the arguments provided in forward func by user can be
        # used to set the model input. Overwrites the empty list of non_default_args
        # because the forward function defined by affine has no non-default arguments
        self._non_default_args = get_non_default_args(forward_func_noshift)
```